core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacade.java

```diff
@@ -116,31 +102,33 @@ private void reportCreationError(Exception ex, Uri videoUri) {
     private final MediaPlayer.OnVideoSizeChangedListener internalSizeChangedListener = new MediaPlayer.OnVideoSizeChangedListener() {
         @Override
         public void onVideoSizeChanged(MediaPlayer mp, int width, int height) {
-            if (onSizeChangedListener == null) {
-                return;
+            if (onVideoSizeChangedForwarder == null) {
+                throw new IllegalStateException("Should bind a OnVideoSizeChangedListener. Cannot forward events.");
```

Comment:
@Dorvaryn good question. 

I had the same debate with @Mecharyry. So basically, in regards to our clients at the moment it is irrelevant if they set the listeners or not. In case they don't the `forwarder` will simply contain an `empty` `Set` of `Listeners`. 

The check is more "internal" for the library. You can consider it a `DeveloperError` and in case reaching that line on the program trace it would mean at some point, somewhere, (concretely `AndroidMediaPlayerImpl`) we forgot to bind to the `forwarder`.